src/sentry/releasehealth/metrics.py

```diff
@@ -0,0 +1,152 @@
+from datetime import datetime
+from typing import Dict, Optional, Sequence, Set
+
+from snuba_sdk import Column, Condition, Entity, Op, Query
+from snuba_sdk.expressions import Granularity
+
+from sentry.models.project import Project
+from sentry.releasehealth.base import ReleaseHealthBackend
+from sentry.sentry_metrics import indexer
+from sentry.sentry_metrics.indexer.base import UseCase
+from sentry.snuba.dataset import Dataset
+from sentry.utils.snuba import raw_snql_query
+
+
+def metric_id(org_id: int, name: str) -> int:
+    index = indexer.resolve(org_id, UseCase.TAG_KEY, name)  # type: ignore
```

Comment:
@untitaker This is what I meant by places where `# type: ignore` can be avoided. Seems to me that indexer is of type `StringIndexer` , and `resolve` always returns `int`
typing the indexer imported should be enough to eliminate a few `# type: ignore`

Reply:
`indexer` is a module. This is a problem of our lazy service abstraction which does too much magic for mypy to comprehend: https://github.com/getsentry/sentry/blob/231349081cb202e82179233750f03905b850a627/src/sentry/sentry_metrics/indexer/__init__.py#L12

We're certainly not going to type the indexer. That module is under active development.

Reply:
What I meant by typing the indexer is  `string_indexer: StringIndexer = indexer` in this class given that this uses an instance of `StringIndexer`, no? and `StringIndexer.resolve` returns an int

Reply:
right but I believe we need a more generic solution to the service problem, I played around with this in hackweek and didn't come up with something satisfactory.

Reply:
Just to be completely clear on this, I do believe some cases do require a `# type: ignore` but what I am advocating here is that we investigate prior to falling back on this 
Imo type casting the instance of `indexer` imported here is a satisfactory alternative to ignoring the type

Reply:
I understand, but we have investigated solutions to give service methods the correct types, I just don't agree with your suggestion as I don't think the time is well-spent in coming up with usage-specific workarounds (per usage of indexer)